src/clib/pio_internal.h

```diff
@@ -76,8 +76,10 @@ extern "C" {
     int  pio_add_to_iodesc_list(io_desc_t *iodesc);
     io_desc_t *pio_get_iodesc_from_id(int ioid);
     int pio_delete_iodesc_from_list(int ioid);
+    int pio_num_iosystem(int *niosysid);
 
     file_desc_t *pio_get_file_from_id(int ncid);
+    int pio_get_file_from_id2(int ncid, file_desc_t **filep);
```

Comment:
How is pio_get_file_from_id2() different from pio_get_file_from_id() and is the new function used in the code?